boards/arm/nrf5340_audio_dk_nrf5340/nrf5340_audio_dk_cpunet_reset.c

```diff
@@ -0,0 +1,164 @@
+/*
+ * Copyright (c) 2020 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr.h>
+#include <init.h>
+#include <logging/log.h>
+
+#include <soc.h>
+#include <drivers/gpio.h>
+
+#include "nrfx_gpiote.h"
+
+LOG_MODULE_REGISTER(nrf5340_audio_dk_nrf5340_cpuapp, CONFIG_LOG_DEFAULT_LEVEL);
+
+#if !defined(CONFIG_TRUSTED_EXECUTION_NONSECURE)
+
+/* This should come from DTS, possibly an overlay. */
+#if defined(CONFIG_BOARD_NRF5340_AUDIO_DK_NRF5340_CPUAPP)
+#define CPUNET_UARTE_PIN_TX 9
+#define CPUNET_UARTE_PIN_RX 8
+#define CPUNET_UARTE_PORT_TRX NRF_P1
+#define CPUNET_UARTE_PIN_RTS 11
+#define CPUNET_UARTE_PIN_CTS 10
+#endif
+
+#if defined(CONFIG_BT_CTLR_DEBUG_PINS_CPUAPP)
+#include <../subsys/bluetooth/controller/ll_sw/nordic/hal/nrf5/debug.h>
+#else
+#define DEBUG_SETUP()
+#endif
+
+#define SHARED_SPI DT_NODELABEL(spi4) /* SD card and HW codec share the SPI4 */
+
+static int core_config(void)
+{
+	int ret;
+	const struct device *gpio_dev;
+	nrf_gpiote_latency_t latency;
+
+	latency = nrfx_gpiote_latency_get();
+
+	if (latency != NRF_GPIOTE_LATENCY_LOWPOWER) {
+		LOG_DBG("Setting gpiote latency to low power");
+		nrfx_gpiote_latency_set(NRF_GPIOTE_LATENCY_LOWPOWER);
+	}
+
+	/* Workaround for issue with PCA10121 v0.7.0 related to SD-card */
+	gpio_dev = device_get_binding(DT_SPI_DEV_CS_GPIOS_LABEL(DT_NODELABEL(sdhc0)));
+	if (!gpio_dev) {
+		return -ENODEV;
+	}
+
+	ret = gpio_pin_configure(gpio_dev, DT_PROP(SHARED_SPI, mosi_pin),
+				 GPIO_DS_ALT_HIGH | GPIO_DS_ALT_LOW);
+	if (ret) {
+		return ret;
+	}
+	ret = gpio_pin_configure(gpio_dev, DT_PROP(SHARED_SPI, sck_pin),
+				 GPIO_DS_ALT_HIGH | GPIO_DS_ALT_LOW);
+	if (ret) {
+		return ret;
+	}
+
+	gpio_dev = device_get_binding("GPIO_0");
```

Comment:
You should get rid of all these hard-coded device names: "GPIO_0", "GPIO_1", "I2C_1", and "SPI_4", and instead query devicetree with suitable `DT_*` macros. Also `DEVICE_DT_GET()` should be used instead of `device_get_binding()` wherever possible.